pkg/build/controller/controller_test.go

```diff
@@ -392,9 +392,6 @@ func TestHandleBuild(t *testing.T) {
 		}
 		if len(tc.outputSpec) != 0 {
 			build := ctrl.BuildStrategy.(*okStrategy).build
-			if build.Spec.Output.To.Name != tc.outputSpec {
```

Comment:
This test is checking that the `build.Spec.Output.To.Name` matches the expected `tc.outputSpec`, in other words, it is checking that the build spec changed... But the build spec should be immutable.

@PI-Victor what we could to here is pass a copy of `build` to `ctrl.HandleBuild(build)` (so to have a pointer to a build that we can guarantee was never shared and never mutated), and then check that the spec of the `build` we passed to `HandleBuild` was not mutated:

```go
if !reflect.DeepEqual(originalBuild.Spec, build.Spec) {
   // error
}}
```